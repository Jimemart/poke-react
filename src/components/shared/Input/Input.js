import React from "react";
import { InputStyled } from "../../../UI";

export const Input = props => {
    const { onInputChange, id, placeholder } = props;
    return (
        <InputStyled
            type="number"
            onChange={onInputChange}
            value={id}
            placeholder={placeholder}
        />
    );
};
