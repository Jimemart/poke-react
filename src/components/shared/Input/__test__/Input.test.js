import React from "react";
import renderer from "react-test-renderer";
import { Input } from "../Input";

describe("Input Component", () => {
    const props = {
        id: 1,
        placeholder: "foo",
        onChange: jest.fn()
    };

    it("should render the Input Component", () => {
        const tree = renderer.create(
            <Input {...props}>
                <div>Some nested content</div>
            </Input>
        );

        expect(tree).toMatchSnapshot("snapshot_1");
    });
});
