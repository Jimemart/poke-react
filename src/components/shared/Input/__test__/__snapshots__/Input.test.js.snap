// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot_1 1`] = `
<input
  className="sc-bxivhb isuOqO"
  onChange={undefined}
  placeholder="foo"
  type="number"
  value=""
/>
`;
